mir-ci/mir_ci/program.py

```diff
@@ -82,27 +96,40 @@ async def kill(self, timeout=default_term_timeout) -> None:
             pass
 
     async def __aenter__(self) -> 'Program':
+        command = self.command
+        if self.with_systemd_run is True:
+            slice = f"mirci-{uuid.uuid4()}"
+            prefix = ("systemd-run", "--user", "--quiet", "--scope", f"--slice={slice}")
+            command = (*prefix, *command)
         process = await asyncio.create_subprocess_exec(
-            *self.command,
+            *command,
             env=dict(os.environ, **self.env),
             stdout=asyncio.subprocess.PIPE,
             stderr=asyncio.subprocess.STDOUT,
             close_fds=True,
             preexec_fn=os.setsid)
         # Without setsid killing the subprocess doesn't kill the whole process tree,
         # see https://pymotw.com/2/subprocess/#process-groups-sessions
-        # Without setsid killing the subprocess doesn't kill the whole process tree,
-        # see https://pymotw.com/2/subprocess/#process-groups-sessions
         async def communicate() -> None:
             raw_output, _ = await process.communicate()
             if self.send_signals_task is not None and not self.send_signals_task.done():
                 self.send_signals_task.cancel()
             self.output = raw_output.decode('utf-8').strip()
         self.process = process
         self.process_end = communicate()
+
+        # If a process runs and dies too quickly, the cgroup file may have never been created
+        try:
+            self.cgroups_task = Cgroup.create(process.pid)
+        except FileNotFoundError as ex:
+            logger.warning(f"Unable to get cgroup for process: {ex}")
```

Comment:
Please add a test covering this.

This will never except, though, will it? Since nothing awaits on the task yet? You should instead catch it, and probably re-raise, in `get_cgroup`?

Reply:
This may be unnecessary, retesting without it in latest